src/seeds.onboarding.cpp

```diff
@@ -255,6 +278,31 @@ void onboarding::_invite(name sponsor, name referrer, asset transfer_quantity, a
 
   uint64_t key = invites.available_primary_key();
 
+  if (campaign_id != 0) {
+    auto citr = campaigns.find(campaign_id);
+    check(citr != campaigns.end(), "campaign not found");
+
```

Comment:
Maybe we should add the future reward amount here too? Otherwise I could create 1000 invites, use up all the funds for that, then run out of rewards, or exceed the balance. 

Ok given that all balances may be mixed up, I think we need to include reward amount here.

Reply:
```
total_quantity = total_quantity + citr -> reward_amount; // something like that
```

Reply:
Example, invite is 20 seeds total, but 10 seeds reward, we need to calculate the total cost of the invite as 30 Seeds, then compare 30 to remaining balance and make sure there's 30 left, then remove 30 already here (?) but not inside the send reward function.... 

Or maybe there's another solution to keep these balances separate. Maybe the rewards are kept track of separately?

Reply:
Yes, you are right, I will correct this